feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,336 @@
+	fptest.LogQuery(t, "Node LLDP", statePath.State(), state)
+
+	if lldpEnabled != state.GetEnabled() {
+		t.Errorf("LLDP enabled got: %t, want: %t.", state.GetEnabled(), lldpEnabled)
+	}
+	if state.GetChassisId() != "" {
+		t.Logf("LLDP ChassisId got: %s", state.GetChassisId())
+	} else {
+		t.Errorf("LLDP chassisID is not proper, got %s", state.GetChassisId())
+	}
+	if state.GetChassisIdType() != 0 {
+		t.Logf("LLDP ChassisIdType got: %s", state.GetChassisIdType())
+	} else {
+		t.Errorf("LLDP chassisIdType is not proper, got %s", state.GetChassisIdType())
+	}
+	if state.GetSystemName() != "" {
+		t.Logf("LLDP SystemName got: %s", state.GetSystemName())
+	} else {
+		t.Errorf("LLDP SystemName is not proper, got %s", state.GetSystemName())
+	}
+
+	got := state.GetInterface(port.Name()).GetName()
+	want := conf.GetInterface(port.Name()).GetName()
+
+	if lldpEnabled && got != want {
+		t.Errorf("LLDP interfaces/interface/state/name = %s, want %s", got, want)
+	}
+}
+
+// checkOtgLldpMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
+func checkOtgLldpMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
+	for _, lldp := range c.Lldp().Items() {
+		lastValue, ok := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameOut().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			txPackets, _ := v.Val()
+			return v.IsPresent() && txPackets != 0
+		}).Await(t)
+		if !ok {
+			txPackets, _ := lastValue.Val()
+			t.Errorf("LLDP sent packets got: %v, want: > 0.", txPackets)
+		}
+		framesIn, _ := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameIn().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			return v.IsPresent()
+		}).Await(t)
+		otgutils.LogLLDPMetrics(t, otg, c)
+		if lldpEnabled {
+			if rxPackets, _ := framesIn.Val(); rxPackets == 0 {
+				t.Errorf("LLDP received packets got: %v, want: > 0.", rxPackets)
+			}
+		} else {
+			if rxPackets, _ := framesIn.Val(); rxPackets != 0 {
+				t.Errorf("LLDP received packets got: %v, want: 0.", rxPackets)
+			}
+		}
+	}
+}
+
+// checkOtgLldpMetrics verifies OTG side lldp neighbor states
+func checkOtgLldpNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) {
+	otgutils.LogLLDPNeighborStates(t, otg, c)
+
+	for lldp, lldpNeighbors := range expLldpNeighbors {
+		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldp).State())
+
+		if len(lldpNeighbors) == 0 {
+			if lldpState.LldpNeighborDatabase != nil {
+				t.Errorf("LLDP Neighbordatabase not expected nil")
+				return
+			}
+		} else {
+			neighbors := lldpState.GetLldpNeighborDatabase().LldpNeighbor
+			for _, lldpNeighbor := range lldpNeighbors {
+				neighborFound := false
+				for _, neighbor := range neighbors {
+					if neighbor.GetChassisId() == lldpNeighbor.chassisId {
```

Comment:
Can we limit the nesting in these if statements?

A couple options:
1) Flatten the logic with "continue" statements
if neighbor.GetChassisId() != lldpNeighbor.chassisId {
  continue
}
if neighbor.GetChassisIdType() != lldpNeighbor.chassisIdType {
  continue
}
...
2) Compare structs in one line with "cmp" or similar library.
Example usage: https://github.com/openconfig/featureprofiles/blob/8072874c8fb62a8313470b220e00f039a57fe016/feature/interface/singleton/ate_tests/singleton_test/singleton_test.go#L251
IIRC there's an option to exclude fields, if you don't need to compare all of them.

Reply:
Added all the checks into one condition